lib/OpenQA/Utils.pm

```diff
@@ -558,5 +560,19 @@ sub job_groups_and_parents {
     return \@res;
 }
 
+sub wait_with_progress {
+    my $interval = ceil(shift);
+    my $tics;
+    local $| = 1;
+
+    do {
```

Comment:
now I'm puzzled. this is just a for loop implemented without using the for keyword, no?